tests/src/test/scala/com/nvidia/spark/rapids/DistributionSuite.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import org.apache.spark.SparkConf
+import org.apache.spark.sql.catalyst.plans.physical.{BroadcastDistribution, ClusteredDistribution, UnspecifiedDistribution}
+
+class DistributionSuite extends SparkQueryCompareTestSuite {
+
+  val seq1 = Seq((10, 20), (10, 30), (20, 20), (20, 30), (30, 20))
+  val seq2 = Seq((10, 20), (20, 300), (30, 400), (10, 400))
+
+  test("BroadcastDistribution and UnknownDistribution") {
+    val conf = new SparkConf()
+    withGpuSparkSession(spark => {
+      import spark.implicits._
+      val df1 = seq1.toDF("a", "b")
+      val df2 = seq2.toDF("aa", "bb")
+      val df = df1.join(df2, df1("a") === df2("aa"), "inner")
+      df.collect()
+
+      val distributions = df.queryExecution.sparkPlan.requiredChildDistribution
+
+      assert(distributions(0).isInstanceOf[UnspecifiedDistribution.type], true)
+      assert(distributions(1).isInstanceOf[BroadcastDistribution], true)
```

Comment:
> the test should fabricate a Catalyst physical plan node with some unknown child distributions, generate a meta that would like to translate it

I am having hard time coming up with the test as described above. Could you please let me know how we can assign unknown child distributions to a physical plan node. For example: If I want to test it for BHJ then instead of BroadcastDistribution how do I replace with something which is unrecognized distribution.

Reply:
Unfortunately we cannot create a brand new top-level distribution since `Distribution` is a sealed trait, but we can create a custom broadcast mode to verify we're checking it properly.  Something similar to:
```scala
case class MyCustomExec() extends ShimSparkPlan {
  override protected def doExecute(): RDD[InternalRow] =
    throw new IllegalStateException("should not be here")
  override def output: Seq[Attribute] = Nil
  override def children: Seq[SparkPlan] = Nil
  override def requiredChildDistribution: Seq[Distribution] =
    Seq(BroadcastDistribution(customBroadcastMode))

  private val customBroadcastMode = new BroadcastMode {
    override def transform(rows: Array[InternalRow]): Any = rows
    override def transform(rows: Iterator[InternalRow], sizeHint: Option[Long]): Any = null
    override def canonicalized: BroadcastMode = this
  }
}

class MyCustomExecMeta(
    plan: RapidsDeltaWriteExec,
    conf: RapidsConf,
    parent: Option[RapidsMeta[_, _, _]],
    rule: DataFromReplacementRule)
    extends SparkPlanMeta[RapidsDeltaWriteExec](plan, conf, parent, rule) {
  override val childPlans: Seq[SparkPlanMeta[SparkPlan]] = Nil
  override val childExprs: Seq[BaseExprMeta[_]] = Nil
  override val childScans: Seq[ScanMeta[_]] = Nil
  override val childParts: Seq[PartMeta[_]] = Nil
  override val childDataWriteCmds: Seq[DataWritingCommandMeta[_]] = Nil
  override def convertToGpu(): GpuExec =
    throw new IllegalStateException("should not be called")
}

[...]

val execRule = GpuOverrides.exec[MyCustomExec](....)
val meta = execRule.wrap(new MyCustomExec)
meta.tagFoGpu()
assert(!meta.canThisBeReplaced)
assert(meta.willWorkOnGpuInfo.contains(...))
```

Reply:
Thank you for providing the example for testing this.  Included this in the PR.